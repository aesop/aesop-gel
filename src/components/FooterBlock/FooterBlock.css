@import 'styles/index';

.base {
  font-family: var(--font-family-sans-serif-regular);

  &.dark {
    color: var(--color-dark-copy);
  }

  &.light {
    color: var(--color-light-copy);
  }
}

.heading {
  margin-top: 0;
  margin-bottom: 0;
  font-family: var(--font-family-sans-serif-medium);
  font-size: rem(15px);
  line-height: 2;
  @media (--viewport-sm-only) {
    font-size: rem(13px);
  }
}

.divider {
  border-bottom: 1px solid var(--color-light-copy);
  margin: 16px 0;

  &.dark {
    border-color: var(--color-dark-copy);
  }

  &.light {
    border-color: var(--color-light-copy);
  }

  @media (--viewport-sm-only) {
    margin: 6px 0;
  }
}

.list {
  padding-left: 0;
  margin: 0;
  list-style: none;
}

.copy,
.listItem {
  font-size: rem(14px);
  line-height: 2;
  @media (--viewport-sm-only) {
    font-size: rem(12px);
  }
  & svg {
    vertical-align: middle;
  }
}

.copy {
  & > :last-child {
    margin-bottom: 0;
  }
}

.learnMore {
  text-decoration: underline;
}
