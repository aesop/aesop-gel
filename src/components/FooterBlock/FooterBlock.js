import React from 'react';
import PropTypes from 'prop-types';
import Hyperlink from '~/components/Hyperlink';
import { HYPERLINK_STYLE_TYPES } from '~/constants';
import cx from 'classnames';
import styles from './FooterBlock.css';

const {
  EXTERNAL_TEXT_LINK,
  INTERNAL_NO_ICON_TEXT_LINK,
} = HYPERLINK_STYLE_TYPES;

const FooterBlock = ({
  className,
  copy,
  isVisibleOnTabletAndMobile,
  links,
  theme,
  title,
}) => {
  const baseClassSet = cx(
    styles.base,
    styles[theme],
    links && !isVisibleOnTabletAndMobile && styles.invisible,
    className,
  );
  const titleClassSet = cx(styles.title);
  const dividerClassSet = cx(styles.divider, styles[theme]);
  return (
    <section className={baseClassSet}>
      <div className={titleClassSet}>{title}</div>
      <div className={dividerClassSet} />
      {links &&
        links.map(link => (
          <div key={link.id}>
            <Hyperlink
              className={styles.item}
              dataTestRef={`DATA_TEST_REF_FOOTER_BLOCK_${link.id}`}
              hasTargetInNewWindow={false}
              style={
                link.openInANewWindow
                  ? EXTERNAL_TEXT_LINK
                  : INTERNAL_NO_ICON_TEXT_LINK
              }
              theme={theme}
              title={link.text}
              url={link.url}
            >
              {link.text}
            </Hyperlink>
          </div>
        ))}
      {copy && <div className={styles.item}>{copy}</div>}
    </section>
  );
};

FooterBlock.propTypes = {
  className: PropTypes.string,
  copy: PropTypes.string,
  isVisibleOnTabletAndMobile: PropTypes.bool,
  links: PropTypes.array,
  theme: PropTypes.oneOf(['dark', 'light']),
  title: PropTypes.string,
};

FooterBlock.defaultProps = {
  className: undefined,
  copy: undefined,
  isVisibleOnTabletAndMobile: false,
  links: undefined,
  theme: 'dark',
  title: undefined,
};

export default FooterBlock;
