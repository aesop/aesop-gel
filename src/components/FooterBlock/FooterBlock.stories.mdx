import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import FooterBlock from './FooterBlock';
import FooterBlockFixture from './FooterBlock.fixture'

# FooterBlock

<Meta title="Components/FooterBlock" component={FooterBlock} />

<Canvas>
  <Story name="Base component">
    <FooterBlock
      copy={FooterBlockFixture.blockNoLink.copy}
      heading={FooterBlockFixture.blockNoLink.heading}
      theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
    />
  </Story>
</Canvas>

## API

<ArgsTable of={FooterBlock} />

<Canvas>
  <Story name="Block Navigation">
    <FooterBlock
      isVisibleOnTabletAndMobile={knobs.boolean('isVisibleOnTabletAndMobile', true, 'Presentation')}
      links={FooterBlockFixture.navigation.links}
      heading={FooterBlockFixture.navigation.heading}
      theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
    />
  </Story>
</Canvas>


