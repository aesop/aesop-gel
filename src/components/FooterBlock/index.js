/* istanbul ignore file */
import FooterBlock from './FooterBlock';

export default FooterBlock;
