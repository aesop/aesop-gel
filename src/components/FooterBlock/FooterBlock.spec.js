import React from 'react';
import renderer from 'react-test-renderer';
import FooterBlock from './FooterBlock';
import FooterBlockFixture from './FooterBlock.fixture';

describe('<FooterBlock />', () => {
  it('should be defined', () => {
    expect(FooterBlock).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(
        <FooterBlock
          copy={FooterBlockFixture.blockNoLink.copy}
          theme="dark"
          title={FooterBlockFixture.blockNoLink.title}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders block navigation component correctly', () => {
    const tree = renderer
      .create(
        <FooterBlock
          isVisibleOnTabletAndMobile={false}
          links={FooterBlockFixture.navigation.links}
          theme="dark"
          title={FooterBlockFixture.navigation.title}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
