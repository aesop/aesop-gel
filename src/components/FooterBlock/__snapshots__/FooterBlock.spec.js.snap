// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterBlock /> should render base component correctly 1`] = `
<div>
  <section
    class="base dark"
  >
    <h1
      class="heading"
    />
    <div
      class="divider dark"
    />
    <div
      class="listItem copy"
    >
      Aesop acknowledges the Traditional Owners of the lands on which we live and work, and pays deepest respects to Elders past, present and emerging. We also acknowledge that sovereignty of these lands has never been ceded.
    </div>
  </section>
</div>
`;

exports[`<FooterBlock /> renders block navigation component correctly 1`] = `
<div>
  <section
    class="base dark"
  >
    <h1
      class="heading"
    />
    <div
      class="divider dark"
    />
    <ul
      class="list"
    >
      <li>
        <a
          class="base hasIcon hasExternalIcon left dark listItem"
          data-test-ref="FOOTER_LINK"
          href="https://www.aesop.com/au/r/contact-us"
          target="_blank"
          title="Contact us"
        >
          Contact us
           
          <i
            aria-hidden="true"
            class="icon"
          >
            <svg
              class="base rightUpArrow dark"
              data-ref=""
              data-testid="data-testid-Icon"
              focusable="false"
              height="12"
              role="img"
              style="height: 12px; width: 12px;"
              tabindex="-1"
              viewBox="0 0 50 50"
              width="12"
            >
              <g>
                <path
                  d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                />
              </g>
            </svg>
          </i>
        </a>
      </li>
      <li>
        <a
          class="base hasIcon hasExternalIcon left dark listItem"
          data-test-ref="FOOTER_LINK"
          href="https://www.aesop.com/au/r/shipping/"
          target="_blank"
          title="Delivery and returns"
        >
          Delivery and returns
           
          <i
            aria-hidden="true"
            class="icon"
          >
            <svg
              class="base rightUpArrow dark"
              data-ref=""
              data-testid="data-testid-Icon"
              focusable="false"
              height="12"
              role="img"
              style="height: 12px; width: 12px;"
              tabindex="-1"
              viewBox="0 0 50 50"
              width="12"
            >
              <g>
                <path
                  d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                />
              </g>
            </svg>
          </i>
        </a>
      </li>
      <li>
        <a
          class="base hasIcon hasExternalIcon left dark listItem"
          data-test-ref="FOOTER_LINK"
          href="https://www.aesop.com/au/r/contact-us"
          target="_blank"
          title="Click And Collect"
        >
          Click And Collect
           
          <i
            aria-hidden="true"
            class="icon"
          >
            <svg
              class="base rightUpArrow dark"
              data-ref=""
              data-testid="data-testid-Icon"
              focusable="false"
              height="12"
              role="img"
              style="height: 12px; width: 12px;"
              tabindex="-1"
              viewBox="0 0 50 50"
              width="12"
            >
              <g>
                <path
                  d="M21.8,0.2l28.1-0.1v28.1l-4.5,0l0.1-20.5L16.2,37.1l-3.4-3.4L42.1,4.4L21.8,4.4L21.8,0.2L21.8,0.2z"
                />
              </g>
            </svg>
          </i>
        </a>
      </li>
      <li>
        <a
          class="base left dark listItem"
          data-test-ref="FOOTER_LINK"
          href="https://www.aesop.com/au/r/shipping/"
          target="_self"
          title="Track Your Orders"
        >
          Track Your Orders
        </a>
      </li>
    </ul>
  </section>
</div>
`;
